.github/workflows/pull-request-ci.yaml

```diff
@@ -199,14 +199,22 @@ jobs:
           echo "::set-env name=SSH_AUTH_SOCK::${SSH_AUTH_SOCK}"
           echo "::set-env name=SSH_AGENT_PID::${SSH_AGENT_PID}"
 
-      # set GKE secret. TBD: use vault instead
+      - name: Query vault for GKE secrets
+        id: vault
+        uses: hashicorp/vault-action@v2.0.1
+        with:
+          exportEnv: false
+          url: https://volt.cap.explore.suse.dev/
+          token: ${{ secrets.VAULT_TOKEN }}
+          secrets: |
+            secret/data/cloud-providers "gke.GKE_CRED_JSON" | GKE_CRED_JSON
```

Comment:
Hmm, when trying this out on elsewhere, it seems like I have to _not_ quote the key:
```suggestion
            secret/data/cloud-providers gke.GKE_CRED_JSON | GKE_CRED_JSON
```
YMMV, I have no idea what I'm doing.

Reply:
Weird. It might be something with JSONata (that is the jq-like library/package used to process the key value).
If your key contains `-` for example (dash, minus) the quoting is required to prevent mishandling of this char as operator instead of part of key.

That said, is your `elsewhere` another GH action, or is it the vault cli ?

Reply:
The `elsewhere` is #1592.

Reply:
Accepted suggestion. Lets see how that goes.